feat(i18n): add Italian locale (it)

Startype-design opened the pull request:
This PR introduces the initial Italian locale (it) and improves the overall i18n structure of the app.

## What’s included

- Added config/locales/it.yml based on the EN locale, with an initial set of translated strings.
- The Italian translation is not yet complete, but covers the main UI sections (auth forms, navigation, dashboard, basic components).
- Identified and added several missing translation keys that were previously hard-coded in the views or components.
These new keys were added to the EN locale and mirrored in the IT file to keep the i18n structure consistent.
- Updated config.i18n.available_locales to include :it.
- Added Italiano to the language selector so users can switch to the new locale.
- Performed a basic review of the UI using the Italian locale to verify that the new keys load correctly.

## Notes / Known Issues

- Some translations are still incomplete. This PR aims to establish the foundation and provide full key coverage; further PRs will refine the Italian language quality.
- Breadcrumbs are not always loading the correct translated values.
This appears to be related to the order in which breadcrumbs are initialized, as some keys resolve before the locale is fully loaded.
- A follow-up investigation is needed to ensure breadcrumb labels consistently pull from the correct i18n namespace.

## Next steps

- Complete the remaining Italian translations in incremental PRs.
- Standardize breadcrumb creation to ensure deterministic loading of i18n keys.
- Continue searching for hard-coded strings and migrate them into locale files.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added comprehensive Italian language support across the app (many UI and email locales).
  * Reports UI now initializes collapsible/reorderable sections.

* **Refactor**
  * Converted many static UI labels and breadcrumbs to translations for consistent localization (navigation, dashboard, transactions, settings, balance sheet, budget categories, etc.).
  * Periods now use localized labels and a new "last_month" period was added.

* **Chores**
  * Marked Italian as a supported locale.
  * Added VS Code workspace ignore rule.

* **Style**
  * Bulk-selection label now shows count with correct pluralization.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alessiocappa commented on config/locales/views/imports/en.yml:
```diff
@@ -36,17 +36,17 @@ en:
       show:
         account_mapping_description: Assign all of your imported file's accounts to
-          Maybe's existing accounts.  You can also add new accounts or leave them
+          Sure's existing accounts.  You can also add new accounts or leave them
```
Isn't it better to use `%{product_name}` as in other files?

alessiocappa commented on config/locales/views/merchants/it.yml:
```diff
@@ -0,0 +1,27 @@
+---
+it:
+  family_merchants:
```
This file has wrong translations for merchant. It should be "commericante" instead of "membro".

alessiocappa commented on config/locales/views/reports/it.yml:
```diff
@@ -0,0 +1,123 @@
+        transactions: transazioni
+        percentage: "% del totale"
+      pagination:
+        showing: Mostrando %{count} transazioni
```
I would use `"%{count} transazioni visualizzate"` or `"Visualizzazione di %{count} transazioni"` instead.

alessiocappa commented on config/locales/views/settings/it.yml:
```diff
@@ -0,0 +1,121 @@
+      show:
+        page_title: Sicurezza
+    settings_nav:
+      accounts_label: Account
```
It should be "Conti" instead of "Account"

alessiocappa reviewed (commented):
@Startype-design, thanks a lot for the PR. As an Italian, it’s fantastic to see the app translated into my native language! 😃

It’s definitely a great starting point, even though some translations are still missing as you mentioned in the PR description. I’ve left a few comments for incorrect translations and believe some points highlighted by CodeRabbit are worth checking out, such as files with incorrect locale keys which could be problematic.

If you can double-check the comments and fix those issues, we can merge the PR and refine it in future ones.

Another aspect that feels slightly off is the labels in the groups section of the homepage:

<img width="346" height="889" alt="Image" src="https://github.com/user-attachments/assets/2acd97e5-d998-4f14-9e55-4d865ec2dbf4" />

I think they should be plural instead of singular (e.g. Conti, Investimenti, Immobili, etc.) but I’m not sure if it’s something already manageable.

Startype-design commented:
@jjmata and @alessiocappa I updated the translation and fixed conflicts and errors reported by CodeRabbit.
I’ll try to finish the last translation file in the next few days, if possible.
